swagger/src/main/scala/org/http4s/rho/swagger/TypeBuilder.scala

```diff
@@ -80,6 +80,35 @@ object TypeBuilder {
 
           models ++ generics ++ children
 
+        case tpe@TypeRef(_, sym: Symbol, tpeArgs: List[Type]) if isSumType(sym) =>
```

Comment:
This whole section feels like it could be broken down nicely into two named methods (pretty much where you have the `flatMap` call), and would be much more readable that way.